### PR TITLE
WIP: UPDATE code generation for ros2_control components, docs ADD docker images for test env

### PR DESCRIPTION
**Pull request by mamueluth**
Code generation:
Code generation for controllers compiles, this should be checked though. Has warnings regarding unused parameter in update(...) function. Was not sure how to adjust update function. Besides test file should be checked if correct node time is used.

Docs:
updated getting stared, installation and building, create workspace.

Docker:
added two containers in test/test_container_docker which can be used as test env for ros team workspace

lot of changes are because of pre-commit, which make the pull request hard read (sry) but they are isolated in one commit.

### CODE REVIEW

File: test/test_container_docker/ubuntu_20_04_ros2_multi/bashrc [hunk 1]
```diff
@@ -0,0 +1,103 @@
+# ~/.bashrc: executed by bash(1) for non-login shells.
```
Comment (destogl) [inline test/test_container_docker/ubuntu_20_04_ros2_multi/bashrc:1]: we should add "main" folder for this files and the same files should e only sym-linked for specific versions

**Review by destogl — CHANGES_REQUESTED** [review]
split this into multiple PR. They should be done feature-wise.

File: .gitignore [hunk 2]
```diff
@@ -1 +1,22 @@
+#general
 docs/_build/*
+*.swp
+_build/*
+
+# nixos and direnv specific
+shell.nix
```
[the remark this answers is not in the document]
Comment (mamueluth) [reply .gitignore:7]: what about the docs/venv (.venv)? If people want to change the docs they have to use python and sphinx for preview. if they follow the Readme.md i added they likely going to use pip. Therefor the docs/venv folder is likely created.

File: _team_config/defines.bash [hunk 3]
```diff
@@ -1 +0,0 @@
-
```
[the remark this answers is not in the document]
Comment (mamueluth) [reply _team_config/defines.bash:1]: To be honest i don't know. Probably and editor issue. Also i can't remember opening the file.

File: templates/ros2_control/controller/controller.cpp [hunk 4]
```diff
@@ -150,7 +155,7 @@ CallbackReturn $ClassName$::on_deactivate(const rclcpp_lifecycle::State & /*prev
   return CallbackReturn::SUCCESS;
 }
 
-controller_interface::return_type $ClassName$::update()
+controller_interface::return_type $ClassName$::update(const rclcpp::Time & time, const rclcpp::Duration & period)
```
Comment (destogl) [inline templates/ros2_control/controller/controller.cpp:158]:
```suggestion
controller_interface::return_type $ClassName$::update(const rclcpp::Time & time, const rclcpp::Duration & /*period*/)
```

**Comment by destogl** [comment]
Obsolete PR – spited into multiple ones., so closing.

**Pull request closed without merging**